test/unit/NormalJob.test.ts

```diff
@@ -116,6 +116,38 @@ describe('NormalJob', () => {
 		expect(job.getEnvironment()).toEqual(environment)
 	})
 
+	it('returns a recursively generated list of prerequisites', () => {
+		const environment: JobEnv = Object.freeze({ dockerImage: 'fake:latest' })
+		const job1 = new NormalJob({ target: '1', commands: [], environment })
+		const job2 = new NormalJob({ target: '2', commands: [], environment })
+		const job3 = new NormalJob({ target: '3', commands: [], environment })
+		const job4 = new NormalJob({ target: '4', commands: [], environment })
+		const job5 = new NormalJob({ target: '5', commands: [], environment })
+		const job6 = new NormalJob({ target: '6', commands: [], environment })
+		const prerequisites1 = new Set<Job>([job1, job2, job3])
+		const prerequisites2 = new Set<Job>([job4, job5, job6])
+		const job7 = new NormalJob({
+			target: '7',
+			prerequisiteJobs: prerequisites1,
```

Comment:
Since you don't use `prerequisites1` anywhere else, and the name isn't particularly indicative, I think it'd be more readable if you just replaced it with the `Set`. Same for `prerequisites2` and `prerequisites3`.

Reply:
Fixed